lib/request.ts

```diff
@@ -355,10 +367,19 @@ export const request = (
       }, options.dropRequestAfter);
     }
 
+    logEvent(EventSource.HTTP_REQUEST, EventName.START);
     if (options.body) {
+      if (typeof options.body.pipe === "function") {
```

Comment:
Is this the recommended way to check if the body is streamable?

Reply:
I couldn't find a standard way for this check. Node.js documentation doesn't say anything and [a third-party package](https://github.com/sindresorhus/is-stream) seems to be an overkill.

In the end, I just want to pipe here. So I think checking the `pipe` method makes sense.

Reply:
I believe `body instanceof Stream` should work as a standard way, but current check is fine too

Reply:
`instanceof` is a bit tricky because there can be multiple stream implementations in an application. For example, it's common to use [readable-stream](https://www.npmjs.com/package/readable-stream) but perron is not using it at the moment. Even if perron used it, an application can have multiple versions of it, and we can't rely on `instanceof`.

Reply:
Yup, I checked in node js source code and saw similar checks, all is good. :smile: